### ISSUE TITLE
Variables: location - improve estimation

### ISSUE THREAD

**Issue by zenourn** [CfSOtago/evAnalysis#6]
When the GPS signal is missing for a sample (>30% of time) currently the location (home, not home) is assumed to be same as the last non-missing GPS signal. This imputation procedure could be greatly improved by using a probabilistic model of location taking into account previous and future known locations, time, charging current, etc.

**Comment by dataknut** [comment]
May need to revisit home/not-home coding - @raffertyparker seems to be seeing some fast charging 'at home'?

**Comment by zenourn** [comment]
Yes, issues with GPS data and code that have to do work on.

**Comment by dataknut** [comment]
Would it be better to just use the kW to define home vs not-home - if we assume home charging is < 7 kW would this work? It assumes no fast chargers at home...

**Comment by zenourn** [comment]
Nope. As an example, I use free 3.3 kW destination charges all the time, and also charge at 3.3 kW at home. Most of my not-home charging is at 3.3 kW rather than at rapid chargers. 
This weekend I was also at Hanmer Springs (not-home) charging at 2 kW.

**Comment by dataknut** [comment]
hmm. @raffertyparker maybe for now we should not try to distinguish between home & not-home - just between 'standard' and 'rapid' - unless Daniel can add the inferred Location field back to the data?

**Comment by raffertyparker** [comment]
>for now we should not try to distinguish between home & not-home - just between 'standard' and 'rapid'
Agreed. 
>to do home vs not-home correctly is actually a full model in itself based upon longitudinal GPS measurements with other variables to help inference.
This sounds like too much work for what we would gain from it.
